docs/source/user_guide/types.rst

```diff
@@ -0,0 +1,145 @@
+======================
+Type Support In Koalas
+======================
+
+.. currentmodule:: databricks.koalas
+
+
+This is based on Koalas 1.1.4.
+
+Koalas supports various of types by mapping them to specific types in PySpark internally.
```

Comment:
I wouldn't focus on internal stuff in the user-facing docs. We should better just document user-facing things. Like, what types you get if you convert a Spark DataFrame to Koalas DataFrame.

Reply:
Let's focus on pandas DataFrame <> Koalas DataFrame <> Spark DataFrame instead of documenting internal stuffs

Reply:
Sounds good. Okay I'll address it, thanks!